corehq/util/couch_doc_processor.py

```diff
@@ -94,7 +94,6 @@ def __iter__(self):
 
         # save iteration state without offset to signal completion
         self.state.pop("offset", None)
-        self.state["retry"] = {}
```

Comment:
> In a situation where some docs are not successfully processed do you expect that the entire migration get's re-run?

Not quite sure what you're asking. All retries are processed at the end after the normal iteration has completed. Also, "every retry document will be yielded each time the iterator is stopped and resumed during the retry phase." I think this is all covered in the [`retry` docstring](https://github.com/dimagi/commcare-hq/blob/5a1d7f866b5f62de145364bd91be3042b1db553e/corehq/util/couch_doc_processor.py#L102-L109). If a retry doc cannot be processed due to too many retries or some other problem that prevents the iteration from completing, then it will be left in the retry queue. It is up to the code using `ResumableDocsByTypeIterator` to decide how that situation will be handled.

Not fond of the proposed/possible solution as it complicates the iteration result by exposing details about retries (every yielded value has an extra boolean `is_retry` flag included with it). The retry mechanism is meant to be an optional feature, and it should be able to be completely ignored and should not otherwise complicate the interface if it is not being used.